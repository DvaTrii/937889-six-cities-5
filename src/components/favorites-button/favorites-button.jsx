import React from "react";
import {FavoritesButtonClassPrefix} from "../../const";
import PropTypes from "prop-types";

export const FavoritesButton = ({classPrefix, isBookmark}) => {
  return (
    <button
      className={`${classPrefix}__bookmark-button button
              ${isBookmark && classPrefix + `__bookmark-button--active`}`}
      type="button"
      // onClick={/* func from dispatch*/}
    >
      <svg
        className={`${classPrefix}__bookmark-icon`}
        width={`${classPrefix === FavoritesButtonClassPrefix.MAIN ? `18` : `31`}`}
        height={`${classPrefix === FavoritesButtonClassPrefix.MAIN ? `19` : `33`}`}>
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">{isBookmark ? `In bookmarks` : `To bookmarks`}</span>
    </button>
  );
};

FavoritesButton.propTypes = {
  classPrefix: PropTypes.string.isRequired,
  isBookmark: PropTypes.bool
};
