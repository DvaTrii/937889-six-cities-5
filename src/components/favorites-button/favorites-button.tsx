import * as React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {postOfferToFavorite} from "../../store/data/operations";
import {FavoritesButtonClassPrefix, FavoritesRequestType} from "../../const";

const FavoritesButton = ({classPrefix, isBookmark, offerId, toggleIsBookmarkProperty}) => {
  return (
    <button
      className={`${classPrefix}__bookmark-button button
              ${isBookmark ? classPrefix + `__bookmark-button--active` : ``}`}
      type="button"
      onClick={() => {
        toggleIsBookmarkProperty(offerId,
            isBookmark ?
              FavoritesRequestType.POP_OUT :
              FavoritesRequestType.PUT_IN);
      }}
    >
      <svg
        className={`place-card__bookmark-icon`}
        width={`${classPrefix === FavoritesButtonClassPrefix.MAIN ? `18` : `31`}`}
        height={`${classPrefix === FavoritesButtonClassPrefix.MAIN ? `19` : `33`}`}>
        <use xlinkHref="#icon-bookmark"></use>
      </svg>
      <span className="visually-hidden">{isBookmark ? `In bookmarks` : `To bookmarks`}</span>
    </button>
  );
};

FavoritesButton.propTypes = {
  classPrefix: PropTypes.string.isRequired,
  isBookmark: PropTypes.bool,
  offerId: PropTypes.number.isRequired,
  toggleIsBookmarkProperty: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  toggleIsBookmarkProperty(id, status) {
    dispatch(postOfferToFavorite(id, status));
  }
});

export {FavoritesButton};
export default connect(null, mapDispatchToProps)(FavoritesButton);
