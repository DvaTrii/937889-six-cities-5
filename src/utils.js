export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
