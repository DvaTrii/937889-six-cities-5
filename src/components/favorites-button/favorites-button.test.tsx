import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import {FavoritesButton} from "./favorites-button";
import {noop} from "../../utils";

const mockStore = configureStore([]);

it(`Should FavoritesButton render properly`, () => {

  const classPrefix: string = `place-card`;
  const offerId: number = 23;
  const store = mockStore({});

  const tree = renderer
    .create(
        <Provider store={store}>
          <FavoritesButton
            classPrefix={classPrefix}
            offerId={offerId}
            toggleIsBookmarkProperty={noop}
          />
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
