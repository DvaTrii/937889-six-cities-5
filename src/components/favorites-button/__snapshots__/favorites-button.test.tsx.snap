// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesButton render properly 1`] = `
<button
  className="place-card__bookmark-button button
              "
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      xlinkHref="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    To bookmarks
  </span>
</button>
`;
